src/server/controllers/authentication.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Autentication routes module
+ *
+ * @request  {post} /login
+ * @request  {post} /registration
+ * @request  {post} /token/refresh
+ */
+
+import User from '../models/user';
+import response from '../helpers/errorHandler';
+import { generateTokensPair } from '../helpers/token';
+import { ClientError, TeapotError } from '../helpers/error';
+import { putInRedis, getFromRedis, deleteFromRedis } from '../helpers/redis';
+
+// internal helpers
+
+const generateTokens = async (user) => {
+  const tokens = generateTokensPair(user);
+  const result = await putInRedis(tokens.refresh, user.userId);
+
+  if (!result) throw new TeapotError();
+
+  return tokens;
+};
+
+// actions
+
+export const registerUser = async (req, res) => {
+  const { username, email, password } = req.body;
+  const user = new User({ username, email, password });
+
+  try {
+    const emailTaken = await User.findOne({ email });
+
+    if (emailTaken) {
+      throw new ClientError('User with this email already exists!');
+    }
+
+    const savedUser = await user.save();
+    if (savedUser) {
+      const tokens = await generateTokens({ userId: user._id });
+      return res.status(201).json({ token: tokens });
+    }
+  } catch (error) {
+    return response(res, error.message, 404);
+  }
+
+  return null;
+};
+
+export const loginUser = async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+
+    if (!user) throw new ClientError('User not exist');
```

Comment:
Please add curly bracers to this if statement here

Reply:
added, done